ipn/ipnlocal/local.go

```diff
@@ -4281,6 +4281,20 @@ func (b *LocalBackend) reconfigAppConnectorLocked(nm *netmap.NetworkMap, prefs i
 					b.logf("Unsuccessful Read RouteInfo: ", err)
 				}
 			}
+			// Re-advertise the stored routes, in case stored state got out of
+			// sync with previously advertised routes in prefs.
+			var domainRoutes []netip.Prefix
+			for _, ips := range ri.Domains {
+				for _, ip := range ips {
+					domainRoutes = append(domainRoutes, netip.PrefixFrom(ip, ip.BitLen()))
+				}
+			}
+			// Note: AdvertiseRoute will trim routes that are already
+			// advertised, so if everything is already being advertised this is
+			// a noop.
+			if err := b.AdvertiseRoute(domainRoutes...); err != nil {
```

Comment:
won't this deadlock? mu is already held, but AdvertiseRoute re-acquires it.

Reply:
oops, yes, because `b.Prefs` in `AdvertiseRoute` locks too.
rewrote this to backfill routes at the end of `authReconfigure` where the lock is already released and added a test.
PTAL